### PR TITLE
Yii2: Fixed output buffer level down to zero if exception occurs

### PR DESCRIPTION
**Pull request by bixuehujin**
This PR is also a fix for #1262

### CODE REVIEW

File: src/Codeception/Lib/Connector/Yii2.php [hunk 1]
```diff
@@ -79,7 +80,15 @@ public function doRequest($request)
         $this->statusCode = null;
 
         ob_start();
-        $app->handleRequest($app->getRequest())->send();
+
+        try {
+            $app->handleRequest($app->getRequest())->send();
+        } catch (Exception $e) {
+            // we shouldn't discard existing output as PHPUnit preform output level verification from PHPUnit 4.2.
+            $app->errorHandler->discardExistingOutput = false;
```
Comment (Ragazzo) [inline src/Codeception/Lib/Connector/Yii2.php]: Current `Codeception` does not use `PhpUnit` version 4

**Comment by Ragazzo** [comment]
Great :+1:

**Comment by bixuehujin** [comment]
Tests for php 5.4 failed due to the failing of `WebDriverTest::testOpenAbsoluteUrls`, not sure how this happen

**Comment by samdark** [comment]
@DavertMik any idea about the failure? Doesn't look like it's connected.

**Comment by bixuehujin** [comment]
Just added a new commit

 For exceptions that not related to http, we pass them to Codeception. This can make our debug easier as Codeception will show the stack backtrace for them.

**Comment by bixuehujin** [comment]
Tests for both 5.4 and 5.6 failed this time

**Comment by DavertMik** [comment]
> any idea about the failure? Doesn't look like it's connected.

Yeah, it's Travis :( In this test we are trying to open http://codeception.com url. Sometimes Travis randomly fails at this step. Stupid travis, stupid.

I restarted the build. @samdark please merge it if you think this PR is ok.

**Comment by samdark** [comment]
Will wait for Travis to finish. Code looks logically correct.

**Comment by samdark** [comment]
Thanks!

**Pull request merged**